osde2e/osde2e-wrapper.py

```diff
@@ -97,6 +97,47 @@ def _download_kubeconfig(osde2ectl_cmd,my_path):
         logging.error('Failed to load cluster-id from metadata.json file located on %s, kubeconfig file wont be downloaded' % my_path)
         return 0
 
+def _add_machinepool(osde2ectl_cmd,my_path):
+    try:
+        metadata = json.load(open(my_path + "/metadata.json"))
+    except Exception as err:
+        logging.error(err)
+        logging.error('Failed to load metadata.json file located %s, machinepool %s wont be created' % (my_path, args.machinepool_name))
+        return 1
+    if 'cluster-id' in metadata and metadata['cluster-id'] != "":
+        cluster_id = metadata['cluster-id']
+        logging.info('Checking if ocm tool is available on the system')
+        ocm_cmd = ["ocm", "-h"]
+        logging.debug(ocm_cmd)
+        ocm_process = subprocess.Popen(ocm_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        ocm_stdout,ocm_stderr = ocm_process.communicate()
+        if ocm_process.returncode != 0:
+            logging.error('%s unable to execute -h' % ocm_cmd)
+            logging.error(ocm_stderr.strip().decode("utf-8"))
+        else:
+            logging.info('Creating machinepool %s on %s' % (args.machinepool_name,cluster_id))
+            # ocm create machinepool --cluster=<your cluster ID> --labels="foo=bar,bar=baz" --replicas=3 --instance-type="m5.xlarge" mp-1
+            machinepool_cmd = ["ocm", "create", "machinepool",
+                               "--cluster", cluster_id,
+                               "--instance-type", args.machinepool_flavour,
+                               "--labels", args.machinepool_labels,
+                               "--taints", args.machinepool_taints,
+                               "--replicas", args.machinepool_replicas,
+                               args.machinepool_name]
+            logging.debug(machinepool_cmd)
+            machinepool_process = subprocess.Popen(machinepool_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            machinepool_stdout,machinepool_stderr = machinepool_process.communicate()
+            if machinepool_process.returncode != 0:
+                logging.error('Unable to create machinepool %s on %s' % (args.machinepool_name,cluster_id))
+                logging.error(machinepool_stdout.strip().decode("utf-8"))
+                logging.error(machinepool_stderr.strip().decode("utf-8"))
+            else:
+                logging.info('Created machinepool %s on %s. Waiting 5 minutes for hosts to come up' % (args.machinepool_name,cluster_id))
+                time.sleep(300)
```

Comment:
Not a big fan of the blind sleep for 5 minutes. Depending on how many nodes it may take more/less than 5 minutes. Can we do a check on the cluster to get the [machinepoolname] node count and loop till it matches the replica count?

Reply:
added active waiting for machinepool creation